designsafe/apps/projects/models/agave/base.py

```diff
@@ -263,6 +264,87 @@ def update_archive(project_directory):
         else:
             update_archive(proj_dir)
 
+    def to_dataset_json(self):
+        """
+        Serialize project to json for google dataset search
+        https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/BMNJPS
+        """
+
+        dataset_json = {
+            "@context": "http://schema.org",
+            "@type": "Dataset",
+            "@id": "",
+            "identifier": "",
+            "logo": "https://www.designsafe-ci.org/static/images/nsf-designsafe-logo.014999b259f6.png",
+            "name": self.title,
+            "creator": [
+                {
+                    "name": "",
+                    "affiliation": "",
+                    "@id": "",
+                    "identifier": ""
+                }
+            ],
+            "author": [
+                {
+                    "name": "",
+                    "affiliation": "",
+                    "@id": "",
+                    "identifier": ""
+                }
+            ],
+            "datePublished": self.created,
+            "dateModified": self.to_body_dict()['lastUpdated'],
+            "description": self.description,
+            "keywords": self.keywords.split(','),
+            "license": {
+                "@type": "Dataset",
+                "text": ""
+            },
+            "publisher": {
+                "@type": "Organization",
+                "name": "Designsafe-CI"
+            },
+            "provider": {
+                "@type": "Organization",
+                "name": "Designsafe-CI"
+            },
+            "includedInDataCatalog": {
+                "@type": "Organization",
+                "name": "Designsafe-CI",
+                "url": "https://designsafe-ci.org"
+            },
+        }
+        if self.dois:
+            dataset_json["distribution"] = {
+                    "@type": "DataDownload",
+                    "name": self.to_body_dict()['value']['projectId'] + "_archive.zip",
+                    "fileFormat": "application/zip",
+                    "contentSize": "",
+                    "@id": "",
+                    "identifier": ""
+            }
+
+        if self.dois:
+            dataset_json['@id'] = self.dois[0]
+            dataset_json['identifier'] = self.dois[0]
+        else:
+            related_ents = self.related_entities()
+            logger.debug(related_ents)
+        if getattr(self, 'team_order', False):
+            authors = sorted(self.team_order, key=lambda x: x['order'])
+        else:
+            authors = [{'name': username} for username in [self.pi] + self.co_pis]
+        dataset_json['creator'] = generate_creators(authors)
+        dataset_json['author'] = generate_creators(authors)
+        if BaseESPublication(project_id=self.project_id):
```

Comment:
I'd like to phase out the use of `BaseES` classes because they sometimes fail to wrap documents correctly. We can use the `IndexedPublication classes instead.
```suggestion
        from designsafe.apps.data.models.elasticsearch import IndexedPublication, IndexedPublicationLegacy
        from designsafe.libs.elasticsearch.exceptions import DocumentNotFound
        ...
        try:
            pub = IndexedPublication.from_id(self.project_id)
        except DocumentNotFound:
            pass
         try:
            pub = IndexedPublicationLegacy.from_id(self.project_id)
         except DocumentNotFound:
            pass
```

Reply:
Done!  Thanks for the suggestion.